client/app.zig

```diff
@@ -402,11 +404,15 @@ fn handleRendererCommand(self: *App, renderer_command: renderite.ParsedCommand)
         .RendererInitProgressUpdate => |renderer_init_progress_update| {
             self.game.load_state.phase.phase_index = @intCast(renderer_init_progress_update.phaseIndex);
 
-            var phase = self.game.load_state.phase;
+            var phase = &self.game.load_state.phase;
 
             phase.phase_name.len = try std.unicode.utf16LeToUtf8(&phase.phase_name.buffer, renderer_init_progress_update.phase);
             phase.sub_phase_name.len = try std.unicode.utf16LeToUtf8(&phase.sub_phase_name.buffer, renderer_init_progress_update.subPhase);
 
+            // null terminate strings
+            phase.phase_name.buffer[phase.phase_name.len] = 0;
+            phase.sub_phase_name.buffer[phase.sub_phase_name.len] = 0;
```

Comment:
You need to give yourself a single byte leeway here to fit the null terminator.

```suggestion
            phase.phase_name.len = try std.unicode.utf16LeToUtf8(phase.phase_name.buffer[0..phase.phase_name.buffer.len - 1], renderer_init_progress_update.phase);
            phase.sub_phase_name.len = try std.unicode.utf16LeToUtf8(phase.sub_phase_name.buffer[0..phase.sub_phase.buffer.len - 1], renderer_init_progress_update.subPhase);

            // null terminate strings
            phase.phase_name.buffer[phase.phase_name.len] = 0;
            phase.sub_phase_name.buffer[phase.sub_phase_name.len] = 0;
```